File: dotfiles/nvim/lua/dap_modules/examples/reference-project/python/hello_debug.py
"""Minimal target for exercising dap_modules' debugpy flow.

Run with: bazel run --config=debugpy //python:hello_debug
then <leader>gp in Neovim (see ../README.md).

debugpy (via the `.bazelrc` `debugpy` config's --wait-for-client) pauses
execution until Neovim attaches, so there's no race to catch it running.
"""

import time


def compute_step(counter: int) -> int:
    squared = counter * counter  # <-- set a breakpoint here (<M-b>)
    return squared


def main() -> None:
    counter = 0
    while True:
        result = compute_step(counter)
        print(f"counter={counter} result={result}", flush=True)
        counter += 1
        time.sleep(0.5)


if __name__ == "__main__":
    main()

File: dotfiles/nvim/lua/dap_modules/examples/reference-project/cpp/hello_debug.cc
// Minimal target for exercising dap_modules' local gdbserver flow:
//   bazel run --config=gdbnf //cpp:hello_debug
// then <leader>bC in Neovim (see ../README.md).
//
// Loops forever, slowly, printing each step -- deliberately not CPU-bound so
// there's plenty of time to attach and set a breakpoint before it matters.

#include <chrono>
#include <cstdio>
#include <thread>

namespace {

int compute_step(int counter) {
  int squared = counter * counter;  // <-- set a breakpoint here (<M-b>)
  return squared;
}

}  // namespace

int main() {
  for (int counter = 0;; ++counter) {
    int result = compute_step(counter);
    std::printf("counter=%d result=%d\n", counter, result);
    std::fflush(stdout);
    std::this_thread::sleep_for(std::chrono::milliseconds(500));
  }
}
